### PR TITLE
Refactor dependency between task resources and container

### PR DESCRIPTION
**Pull request by Realmonia**
### Summary

This change does the refactor to remove the existing code dependency between task resources and container. This will allow container to depend on task resources and make it easier for container execution role and secret integration.

### Implementation details

Move related resources and its status class to containerresource package.

### Testing

make test and make release. Further regression tests will be done and attach result here.

New tests cover the changes:  n/a

### Description for the changelog

Refactor the dependency between task resources and container.

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: agent/api/container/container.go [hunk 1]
```diff
@@ -20,6 +20,9 @@ import (
 	"sync"
 	"time"
 
+	"github.com/aws/amazon-ecs-agent/agent/containerresource"
+	"github.com/aws/amazon-ecs-agent/agent/containerresource/containerstatus"
+
```
Comment (mythri-garaga) [inline agent/api/container/container.go:25]: nit: can we remove extra lines here?

File: agent/api/container/container_test.go [hunk 2]
```diff
@@ -21,6 +21,9 @@ import (
 	"testing"
 	"time"
 
+	containerresource "github.com/aws/amazon-ecs-agent/agent/containerresource"
+	"github.com/aws/amazon-ecs-agent/agent/containerresource/containerstatus"
+
```
Comment (mythri-garaga) [inline agent/api/container/container_test.go:26]: nit: extra line here too and other files too

**Review by mythri-garaga — APPROVED** [review]

**Review by angelcar — APPROVED** [review]

File: agent/api/container/container_test.go (hunk 2 above)
Comment (Realmonia) [reply agent/api/container/container_test.go:26]: Will update!

**Comment by Realmonia** [comment]
Update the PR based on the rebase PR and added some missed dependency refactor (moved errors/containerevent/port_bindings/transport from api/container to containerresource)

**Review by mythri-garaga — APPROVED** [review]

**Review by angelcar — APPROVED** [review]

**Pull request merged**